ios/sound-send/sound-send/ViewController.swift

```diff
@@ -9,17 +9,98 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    //the base url, change it when we have one i guess?
+    let baseurl = "http://www.soundsend.com"
+    //keep track of whether a session is in progress. there shouldn't be a session made if one is in progress
+    var sessioninprogress = false
```

Comment:
@bluesnowstorm infact I propose a channel class and an optional variable that points to the current channel. This way we can forgo the need for a state variable and instead rely on the power of optionals. have something like:
```
var currentChannel = Channel(...)
```
When a broadcaster cancels a request (this is sort of thinking ahead) we can call the destructor on currentChannel and set it to nil.